### PR TITLE
spike: Investigate runtime and file queueing

### PR DESCRIPTION
**Pull request by CollectiveUnicorn**
- Adds an ADR to determine how we should go about handling performance bottlenecks, in addition to what tools we should use if any for queuing.
- Closes: https://github.com/defenseunicorns/leapfrogai/issues/746

### CODE REVIEW

File: adr/0006-queueing-high-traffic.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  - [References](#references)
+
+## Status
+
+PROPOSED
+
+## Context
+
+LeapfrogAI has to be able to handle a large volume of inference, file upload, and embeddings requests. To ensure that we can handle this level of activity without significant performance degradation we need to have systems in place to prevent the system from being overwhelmed or blocked by a large volume or a single long running task. 
+
```
Comment (alekst23) [inline adr/0006-queueing-high-traffic.md:22]:
```suggestion
Adding a Queue management component can help us create a more efficient request management system to deal with high request volumes of long-running tasks. However, it introduces a significant level of complexity to the system, and we must weight the options carefully.

### Benefits of Queue system for request processing:
- Allows the Gateway to respond even when the back-end is very busy
- Prevents request from being dropped or timing out
- Allows resuming failed requests
- Allows to throttle message processing rate
```

Comment (CollectiveUnicorn) [reply adr/0006-queueing-high-traffic.md:22]: This is good, though needs to be weigh instead of weight

**Comment by barronstone** [comment]
The rationale makes sense to me at a high level: Take a lightweight approach for now and take the more complex path later if necessary.

**Review by gphorvath — APPROVED** [review]

**Comment by gphorvath** [comment]
This makes sense as `Proposed`. We need to make sure to revisit as we begin implementing.

**Review by gphorvath — APPROVED** [review]

**Pull request merged**